src/api/purchase/use-purchase.tsx

```diff
@@ -0,0 +1,47 @@
+import type { UseMutationOptions } from '@tanstack/react-query';
```

Comment:
just a general thing, if you're not returning a `jsx` component just use `ts` extention intead of `tsx`

Reply:
Done